rtc-video-quality/generate_data.py

```diff
@@ -453,18 +459,29 @@ def generate_jobs(args, temp_dir):
     jobs = []
     for clip in args.clips:
         bitrates = find_bitrates(clip['width'], clip['height'])
-        for bitrate_kbps in bitrates:
+        qp_values = find_qp()
+
+        if args.enable_bitrate:
+            params = bitrates
+        else:
+            params = qp_values
+        for param in params:
             for (encoder, codec) in args.encoders:
                 job = {
+                    'param':
+                        'bitrate' if args.enable_bitrate else 'qp',
                     'encoder':
                         encoder,
                     'codec':
                         codec,
                     'clip':
                         clip,
+                    'qp_value':
+                        param if not args.enable_bitrate else -1,
                     'target_bitrates_kbps':
-                        split_temporal_bitrates_kbps(bitrate_kbps,
-                                                     args.num_temporal_layers),
+                        split_temporal_bitrates_kbps(param,
+                                                     args.num_temporal_layers)
+                        if args.enable_bitrate else [],
```

Comment:
I don't really like seeing so many conditionals here. It would be better to separate the code which is generic from the qp-specific and bitrate-specific parts. https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Maybe set up all the other fields of job, and then do something like this:

if (args.enable_bitrate):
  jobs.update({'param': 'bitrate', 'target_bitrates_kbps': ...})
else:
  jobs.update({'param': 'qp', 'qp_value': ...})